pod/video/utils.py

```diff
@@ -44,12 +45,68 @@
 
 SECURE_SSL_REDIRECT = getattr(settings, 'SECURE_SSL_REDIRECT', False)
 
+FFPROBE = getattr(settings, 'FFPROBE', 'ffprobe')
+
+GET_DURATION_VIDEO = getattr(
+    settings,
+    'GET_DURATION_VIDEO',
+    "%(ffprobe)s -v error -show_entries format=duration "
+    + "-of default=noprint_wrappers=1:nokey=1 -i %(source)s")
 
 # ##########################################################################
 # ENCODE VIDEO : GENERIC FUNCTION
 # ##########################################################################
 
 
+def get_duration_from_mp4(mp4_file, output_dir):
+    msg = ""
+    duration = 0
+    command = GET_DURATION_VIDEO % {'ffprobe': FFPROBE, 'source': mp4_file}
+    ffproberesult = subprocess.run(
+        command, shell=True, stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT)
+    msg += "\nduration ffprobe command: \n- %s\n" % command
+
+    try:
+        duration = int(float("%s" % ffproberesult.stdout.decode('utf-8')))
+    except (RuntimeError, TypeError, AttributeError, ValueError) as err:
+        msg += "\nUnexpected error: {0}".format(err)
+
+    with open(output_dir + "/encoding.log", "a") as f:
+        f.write(msg)
+        f.write('\nffprobe command duration video result: ')
+        f.write(ffproberesult.stdout.decode('utf-8')+"\n")
+        f.write("Duration : %s \n\n" % duration)
+
+    if DEBUG:
+        print(msg)
+        print('\nffprobe command duration video result: ')
+        print(ffproberesult.stdout.decode('utf-8'))
+        print("Duration : %s" % duration)
+
+    return duration
+
+
+def fix_video_duration(video_id, output_dir):
+    vid = Video.objects.get(id=video_id)
```

Comment:
@ptitloup  on devrait pas, ici, prevoir un not found et du coups un crach ? 
Utilisation de filter et la methode first() de celui-ci peut etre ?

Reply:
C'est ok, je viens de faire un commit pour rajouter un try catch